### ISSUE TITLE
Stage 2 training results in dark and saturated images

### ISSUE THREAD

**Issue by man-sean** [sczhou/CodeFormer#407]
I'm trying to reproduce the results of Stage 2, but the images I get are very dark and saturated even after 380K steps:
![image](https://github.com/user-attachments/assets/a0a0e623-292a-4a36-997c-c24156783078)

I'm using the provided Stage 2 config, using the pre-trained Stage 1 checkpoint with the pre-computed latents.

The losses seem to converge, but the images do not get better:
<img width="913" alt="image" src="https://github.com/user-attachments/assets/be8a281a-176c-4609-8bd2-03e15f52a6c0">
This is an example from 10K steps (early in the training):
![image](https://github.com/user-attachments/assets/185a025f-a72e-4aa9-9212-0103c069b86e)

Has anyone encountered this problem?

**Comment by man-sean** [comment]
I found the problem. The [validation code](https://github.com/sczhou/CodeFormer/blob/e878192ee253cfcc8f19e29d3307c181501f53ae/basicsr/models/codeformer_idx_model.py#L158) does not normalize the resulting images back to [0,1] from [-1,1], resulting in severe pixel clipping.

The change:
```python
visuals = self.get_current_visuals()
sr_img = tensor2img([visuals['result']], min_max=[-1,1])
if 'gt' in visuals:
    gt_img = tensor2img([visuals['gt']], min_max=[-1,1])
    del self.gt
```

Result after 500 steps:
![image](https://github.com/user-attachments/assets/d32376ec-eac3-4a43-b67d-98533f937dba)
